sdk/src/server-api/sc/api/plugins/TwoPlayerGameState.kt

```diff
@@ -20,40 +20,36 @@ abstract class TwoPlayerGameState<P : Player>(
     val players: List<P>
         get() = listOf(first, second)
     
+    /** Die Namen der beiden Spieler. */
+    val playerNames: Array<String>
+        get() = arrayOf(first.displayName, second.displayName)
+    
     /** The Team active in the current turn. */
     abstract val currentTeam: ITeam
 
     /** The Player whose team's turn it is. */
     open val currentPlayer: P
-        get() = getPlayer(currentTeam)!!
+        get() = getPlayer(currentTeam)
 
     /** The player opposite to the currently active one. */
     val otherPlayer: P
-        get() = getPlayer(otherTeam)!!
+        get() = getPlayer(otherTeam)
 
     /** The Team opposite to the currently active one. */
     val otherTeam: ITeam
         get() = currentTeam.opponent()
 
     /** Der Spieler, der das Spiel begonnen hat. */
     val startPlayer: P
-        get() = getPlayer(startTeam)!!
-
-    /** Die Namen der beiden Spieler. */
-    val playerNames: Array<String>
-        get() = arrayOf(first.displayName, second.displayName)
+        get() = getPlayer(startTeam)
 
     /** Letzter getaetigter Zug. */
     abstract val lastMove: IMove?
 
     fun getOpponent(player: P) =
             getPlayer(player.color.opponent())
 
-    fun getPlayer(team: ITeam): P? = when(team.index) {
-        0 -> first
-        1 -> second
-        else -> null
-    }
+    fun getPlayer(team: ITeam): P = players[team.index]
```

Comment:
Don't you at least want to throw a proper error?
We can't guarantee that a future plugin will only have exactly two teams

Reply:
well then it will throw an `ArrayIndexOutOfBoundsException`, which seems fitting to me.
Also, the fact that we have two teams is still hardcoded in a lot of places. This is actually one of my attempts to reduce that.
Given the structure of the Software-Challenge, I however don't see a case where we would have more than two players.